grafana-plugin/src/containers/ServiceNowConfigDrawer/ServiceNowAuthSection.tsx

```diff
@@ -0,0 +1,67 @@
+import React, { useState } from 'react';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { Button, HorizontalGroup, Icon, LoadingPlaceholder, useStyles2 } from '@grafana/ui';
+import { observer } from 'mobx-react';
+import { useFormContext } from 'react-hook-form';
+
+import { RenderConditionally } from 'components/RenderConditionally/RenderConditionally';
+import { Text } from 'components/Text/Text';
+import { IntegrationFormFields } from 'containers/IntegrationForm/IntegrationForm';
+import { ApiSchemas } from 'network/oncall-api/api.types';
+import { useCurrentIntegration } from 'pages/integration/OutgoingTab/OutgoingTab.hooks';
+import { useStore } from 'state/useStore';
+import { OmitReadonlyMembers } from 'utils/types';
+
+import { getCommonServiceNowConfigStyles } from './ServiceNow.styles';
+import { ServiceNowFormFields } from './ServiceNowStatusSection';
+
+export const ServiceNowAuthSection: React.FC = observer(() => {
+  const { getValues } = useFormContext<ServiceNowFormFields | IntegrationFormFields>();
+  const { alertReceiveChannelStore } = useStore();
+
+  const currentIntegration = useCurrentIntegration();
+  const [isAuthTestRunning, setIsAuthTestRunning] = useState(false);
+  const [authTestResult, setAuthTestResult] = useState(undefined);
```

Comment:
it seems that the state here is always `boolean`. How about we make it more type safe and we use `useState(false)`? Then TS will infer the type from the default value. Without it `authTestResult` has `any` type

Reply:
Undefined serves its role here because I'm actually checking for undefined and not for false initially `!isAuthTestRunning && authTestResult !== undefined`